lib/cuda/cudaBasebandBeamformer.cpp

```diff
@@ -79,11 +80,10 @@ cudaEvent_t cudaBasebandBeamformer::execute(int gpu_frame_id, cudaEvent_t pre_ev
     int32_t* info_memory =
         (int32_t*)device.get_gpu_memory_array(_gpu_mem_info, gpu_frame_id, info_len);
 
-    if (pre_event)
-        CHECK_CUDA_ERROR(cudaStreamWaitEvent(device.getStream(CUDA_COMPUTE_STREAM), pre_event, 0));
-    CHECK_CUDA_ERROR(cudaEventCreate(&pre_events[gpu_frame_id]));
-    CHECK_CUDA_ERROR(
-        cudaEventRecord(pre_events[gpu_frame_id], device.getStream(CUDA_COMPUTE_STREAM)));
+    if (pre_events[cuda_stream_id])
+        CHECK_CUDA_ERROR(
+            cudaStreamWaitEvent(device.getStream(cuda_stream_id), pre_events[cuda_stream_id], 0));
```

Comment:
I think we can actually remove the wait event on the pre_event.  Since the pre_event is now on the same stream streams, this doesn't provide any meaningful synchronization past what the stream ordering guarantees already.   Before this was used to sync streams, but that's now provided by `syncStream`.

Reply:
Does this go for all cudaCommand compute subclasses?  Eg this same snippet also appears in `cudaCorrelatorAstron.cpp`, `cudaCorrelator.cpp` and `cudaOutputDataZero.cpp`.

Reply:
Yes I think we can remove this from all subclasses (other than the sync of course).

Reply:
Right, I see what you mean now, these were just waiting on the previous event in their stream, which is... already how a stream works :)  Done!